# STATISTICAL INFERENCE: THIRD SESSION

## Hypothesis Testing: One population

This computer practise is related to the previous one, in which we saw estimations and Confidence Intervals. Precisely, we will use the same commands and see how we modify them so as to perform hypothesis testing.

In [8]:
# read heights.txt

Data <-read.table(file="heights.txt", header=TRUE, dec=",", sep="\t")

# For the test of the variances of this session, we will need to install and load the following packet:
# install.packages("EnvStats")
# library(EnvStats)


### Population mean weight 

#### We formulate the following hypothesis test:

 $$H_0:\ \mu = 80$$

 $$H_1:\ \mu \neq 80$$

With a significance level of 1%

In [2]:
t.test(Data$WEIGHT,alternative='two.sided',mu=80)


	One Sample t-test

data:  Data$WEIGHT
t = -5.443, df = 170, p-value = 1.805e-07
alternative hypothesis: true mean is not equal to 80
95 percent confidence interval:
 69.83973 75.24799
sample estimates:
mean of x 
 72.54386 


#### Explanation:

The obtained statistic is -5.44299, the degrees of freedom is 170 and the p-valu 1.805397e-07 (that is, 1.8 times 10 to the power of -7. With a significance level of 1%, we conclude that the mean weight is not 80 (since the p-value is smaller than 0.01).

#### Observations: 

* If instead of 80, we aim to check the value 60, we must modify mu=80 by mu=60.
* For the unilateral right test, we must put alternative="greater" 
* For the unilateral left test, we must put alternative="less" 
* Remark that, unlike in the previous session, the argument conf.level does not appear. Why?

### Hypothesis test of the variance of the heights

 $$H_0:\ \sigma^2=16$$

  $$H_1: \ \sigma^2\neq 16$$
  
We consider a significance level of 5%

In [7]:
varTest(Data$HEIGHT,alternative  = "two.sided",sigma=16)


Results of Hypothesis Test
--------------------------

Null Hypothesis:                 variance = 16

Alternative Hypothesis:          True variance is not equal to 16

Test Name:                       Chi-Squared Test on Variance

Estimated Parameter(s):          variance = 11.08242

Data:                            Data$HEIGHT

Test Statistic:                  Chi-Squared = 117.7507

Test Statistic Parameter:        df = 170

P-value:                         0.001614072

95% Confidence Interval:         LCL =  9.057947
                                 UCL = 13.874455


#### Explanation:

The obtained statistic is 117.7507, the degrees of freedom of the $\chi^2$ is 170 and the p-value is 0.001614. With a significance level of 5%, we conclude that the variance of the heights is not 16 (since the p-value is smaller than 0.05). 

#### Observations:

* If instead of 16, we aim to test the value 9, we must replace sigma=16 by sigma=9.
* For the unilateral right test, we must put alternative="greater" 
* For the unilateral left test, we must put alternative="less" 
* Remark that, unlike in the previous session, the argument conf.level does not appear. Why?

## Hypothesis test of the proportion of those from Alava

$$H_0:\ p\leq 0.15$$

$$H_1:\ p>0.15$$

We consider a significance level of 5%.

In [9]:
# we count how many are from Alava
Data.alava <- subset(Data,BIRTHPLACE==1)
amount.alava <- length(Data.alava$HEIGHT)
#amount.alava
total <- length(Data$HEIGHT)
#total
prop.test(amount.alava,total,p=0.15,alternative="greater")


	1-sample proportions test with continuity correction

data:  amount.alava out of total, null probability 0.15
X-squared = 7.5736, df = 1, p-value = 0.002962
alternative hypothesis: true p is greater than 0.15
95 percent confidence interval:
 0.1771247 1.0000000
sample estimates:
        p 
0.2280702 


#### Explanation:

The value of the estatistic is 7.573558 and of the p-value 0.002961537. The conclusions is that, with a significance level of 5%, the proportion of population from Alava is larger than 0.15 (since the p-value is smaller than 0.05).
 
#### Observations:

* If instead of the value 0.15, we aim to test the value 0.4, we replace p=0.15 by p=0.4.
* For the bilateral test, we must put alternative="two.sided"
* For the unilateral left test, we must put alternative="less" 
* Remark that, unlike in the previous session, the argument conf.level does not appear. Why?